### ISSUE TITLE
Screen orientation lock fail on modern browser

### ISSUE THREAD

**Issue by malcom** [emscripten-core/emscripten#18693]
The HTML5 Emscripten functions around locking the screen orientation, like:

- `emscripten_lock_orientation`
- `emscripten_unlock_orientation`
- `emscripten_get_orientation_status`

based on the [old API](https://developer.mozilla.org/en-US/docs/Web/API/Screen/lockOrientation) (`screen.lockOrientation`), which is now deprecated and removed from modern browsers, and replaced by the new [Screen Orientation API](https://developer.mozilla.org/en-US/docs/Web/API/ScreenOrientation) (`screen.orientation.*`). This causes that this functionality does not work on current version of most available browsers.

It would be nice to add support to the new API in Emscripten.

Anyway, there is also **a problem in the current implementation that can lead to crashes/bugs**. As both APIs use the same name - `screen.orientation`, calling `emscripten_get_orientation_status` in a modern browser will fail and cause a memory error due to access to invalid data - instead of the expected lock type (`string`) it will get ref to object that exposed the new interface (`ScreenOrientation`).

**Comment by M-SAI-SOORYA** [comment]
Yes, you are correct. The Emscripten functions for locking the screen orientation currently rely on the deprecated Screen.lockOrientation() API, which is no longer supported in modern browsers. To address this issue, Emscripten would need to be updated to use the newer Screen Orientation API, which provides a more flexible and robust way to control screen orientation in web applications.

To implement support for the Screen Orientation API in Emscripten, you would need to update the code that handles screen orientation in the Emscripten runtime. This would involve replacing the calls to Screen.lockOrientation() and related functions with calls to the new Screen Orientation API, which provides similar functionality in a more standardized and future-proof way.

As for the issue with the current implementation leading to crashes and bugs, this is a serious concern that would need to be addressed as part of any update to the Emscripten runtime. One possible solution would be to modify the emscripten_get_orientation_status() function to check whether the current browser supports the Screen Orientation API, and if so, use that API to retrieve the current orientation. If not, it could fall back to the deprecated Screen.lockOrientation() API to retrieve the current orientation.

**Comment by ericoporto** [comment]
Does anyone knows how to detect if it worked or not in Safari? Does it generate an exception? In the documentation it appears it's only supported through switches in a preview browser

**Comment by igagis** [comment]
also facing this problem.
